app/controllers/application_controller.rb

```diff
@@ -542,6 +542,16 @@ def handle_permalink(path)
     end
   end
 
+  def rate_limit_second_factor!(user)
+    return if params[:second_factor_token].blank?
+
+    RateLimiter.new(nil, "second-factor-min-#{request.remote_ip}", 3, 1.minute).performed!
+
+    if user
+      RateLimiter.new(nil, "second-factor-min-#{user.username}", 3, 1.minute).performed!
```

Comment:
Not sure if we want to do `RateLimiter.new(nil, "second-factor-min-anonymous", 3, 1.minute).performed!` if the user does not exist?

Reply:
I think this is ok, but we are going to hit this multiple times in a few workflows lets double it from 3 a minute to 6 a minute